fix: erUtils.ts の型エラー (null 非許容)

## 現象

`npm run build` で以下のエラーが発生する。

```
src/utils/erUtils.ts(48,3): error TS2322: Type '(... | null)[]' is not assignable to type 'ErRelation[]'.
src/utils/erUtils.ts(66,23): error TS2677: A type predicate's type must be assignable to its parameter's type.
```

## 原因

`erUtils.ts` 内でnullを含む配列を `ErRelation[]` に代入しようとしている。type guard の型定義にも不整合がある。

## 対応

- `.filter(Boolean)` を適切な型ガードに置き換えるか、`ErRelation | null` → `ErRelation` に絞り込む処理を修正する。

--- a/designer/src/utils/erUtils.ts
+++ b/designer/src/utils/erUtils.ts
@@ -46,7 +46,7 @@ export function convertLogicalRelations(
   const tableIdMap = new Map(tables.map((t) => [t.id, t]));
 
   return logicals
-    .map((lr) => {
+    .map((lr): ErRelation | null => {
       const src = tableIdMap.get(lr.sourceTableId);
       const tgt = tableIdMap.get(lr.targetTableId);
       if (!src || !tgt) return null;
@@ -61,7 +61,7 @@ export function convertLogicalRelations(
         cardinality: lr.cardinality,
         physical: false,
         label: lr.label,
-      } satisfies ErRelation;
+      };
     })
     .filter((r): r is ErRelation => r !== null);
 }

